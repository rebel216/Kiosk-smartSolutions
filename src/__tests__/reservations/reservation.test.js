import { BrowserRouter } from "react-router-dom";
import AddReservation from "../../components/reservations/reservation";
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
    reservation: [],
    vehical: [],
    user: {
        id: 1
    }
  };

const store = mockStore(initialState);

test('Reservation form match snapshot', () => {
    const component = renderer.create(
        <BrowserRouter>
        <Provider store={store}>
            <AddReservation />
                </Provider>
        </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});