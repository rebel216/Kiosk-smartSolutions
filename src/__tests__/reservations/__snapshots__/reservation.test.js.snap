// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation form match snapshot 1`] = `
<div
  className="container "
>
  <span
    className="add-vehicle-header"
  >
     Reserve
  </span>
  <div
    className="row"
  >
    <form
      className="add-vehicle-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          User name 
        </label>
        <input
          className="form-control"
          disabled={true}
          id="formGroupExampleInput"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Date
        </label>
        <input
          className="form-control"
          id="formGroupExampleInput2"
          name="reserve_date"
          onChange={[Function]}
          required={true}
          type="date"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Address
        </label>
        <input
          className="form-control"
          id="formGroupExampleInput2"
          name="address"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Vehicle
        </label>
        <select
          className="form-control"
          id="exampleFormControlSelect1"
          name="vehicle_id"
          onChange={[Function]}
        />
      </div>
      <div
        className="button-green"
      >
        <button
          className="btn btn-primary btn-lg btn-block detail-reserve-btn"
          type="submit"
        >
          Reserve
        </button>
      </div>
    </form>
  </div>
</div>
`;
